File: backend/app/schemas/report.py
"""Report related schemas."""
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CareerReportBase(BaseModel):
    """Base career report schema."""
    title: str | None = None
    summary: str | None = None
    recommendations: list[dict[str, Any]] | None = None
    suggested_jobs: list[dict[str, Any]] | None = None
    skill_gaps: list[dict[str, Any]] | None = None
    career_path: list[dict[str, Any]] | None = None


class CareerReportCreate(CareerReportBase):
    """Career report creation schema."""
    student_id: UUID


class CareerReportUpdate(BaseModel):
    """Career report update schema."""
    title: str | None = None
    summary: str | None = None
    recommendations: list[dict[str, Any]] | None = None
    suggested_jobs: list[dict[str, Any]] | None = None
    skill_gaps: list[dict[str, Any]] | None = None
    career_path: list[dict[str, Any]] | None = None


class CareerReportResponse(CareerReportBase):
    """Career report response schema."""
    id: UUID
    student_id: UUID
    status: str = "pending"
    version: str = "1.0"
    content_json: dict[str, Any] | None = None
    pdf_path: str | None = None
    docx_path: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ReportVersionBase(BaseModel):
    """Base report version schema."""
    version: str
    content: dict[str, Any]
    change_notes: str | None = None


class ReportVersionCreate(ReportVersionBase):
    """Report version creation schema."""
    report_id: UUID


class ReportVersionResponse(ReportVersionBase):
    """Report version response schema."""
    id: UUID
    report_id: UUID
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ReportGenerateRequest(BaseModel):
    """Request schema for generating a report."""
    student_id: UUID
    job_ids: list[UUID] | None = None
    include_export: bool = False


class ReportExportRequest(BaseModel):
    """Request schema for exporting a report."""
    report_id: UUID
    format: str = "pdf"  # pdf, docx

File: frontend/src/types/report.ts
// Report types - aligned with backend/app/schemas/report.ts

export interface CareerReportBase {
  title?: string;
  summary?: string;
  recommendations?: Record<string, unknown>[];
  suggested_jobs?: Record<string, unknown>[];
  skill_gaps?: Record<string, unknown>[];
  career_path?: Record<string, unknown>[];
}

export interface CareerReportCreate extends CareerReportBase {
  student_id: string;
}

export interface CareerReportUpdate {
  title?: string;
  summary?: string;
  recommendations?: Record<string, unknown>[];
  suggested_jobs?: Record<string, unknown>[];
  skill_gaps?: Record<string, unknown>[];
  career_path?: Record<string, unknown>[];
  content_json?: Record<string, unknown>;
}

export interface CareerReportResponse extends CareerReportBase {
  id: string;
  student_id: string;
  pdf_path?: string;
  docx_path?: string;
  version?: string;
  status?: string;
  content_json?: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface ReportDimension {
  key: string;
  label: string;
  score: number;
  reason: string;
}

export interface ReportActionItem {
  priority: string;
  item: string;
  gap_desc: string;
  score_impact: number;
  action: string;
  timeline: string;
}

export interface ReportPathNode {
  stage: string;
  title: string;
  condition: string;
  is_current?: boolean;
}

export interface ReportPaths {
  primary_path: ReportPathNode[];
  alt_paths: { title: string; skill_overlap?: number }[];
}

export interface ReportChapter {
  chapter_id: number;
  title: string;
  text: string;
  data?: unknown;
  status?: string;
}

export interface ReportContent {
  title: string;
  summary: string;
  target_job: Record<string, unknown>;
  dimensions: ReportDimension[];
  actions: ReportActionItem[];
  paths: ReportPaths;
  chapters: ReportChapter[];
  metadata?: Record<string, unknown>;
}

export interface ReportVersionBase {
  version: string;
  content: Record<string, unknown>;
  change_notes?: string;
}

export interface ReportVersionCreate extends ReportVersionBase {
  report_id: string;
}

export interface ReportVersionResponse extends ReportVersionBase {
  id: string;
  report_id: string;
  created_at: string;
}

export interface ReportGenerateRequest {
  student_id: string;
  job_profile_ids?: string[];
  include_export: boolean;
}

export interface ReportExportRequest {
  report_id: string;
  format: 'pdf' | 'docx';
}
